src/index.ts

```diff
@@ -15,24 +15,27 @@ const PARENT_DIRECTORY = /^(\/?\.\.)+/;
 const ESCAPING_BACKSLASHES = /\\(?=[()[\]{}!*+?@|])/g;
 const BACKSLASHES = /\\/g;
 
+// The `Array.isArray` type guard doesn't work for readonly arrays.
+const isReadonlyArray: (arg: unknown) => arg is readonly unknown[] = Array.isArray;
+
 export interface GlobOptions {
-  absolute?: boolean;
-  caseSensitiveMatch?: boolean;
-  cwd?: string;
-  debug?: boolean;
-  deep?: number;
-  dot?: boolean;
-  expandDirectories?: boolean;
-  followSymbolicLinks?: boolean;
-  globstar?: boolean;
-  ignore?: string | string[];
-  onlyDirectories?: boolean;
-  onlyFiles?: boolean;
+  readonly absolute?: boolean;
+  readonly caseSensitiveMatch?: boolean;
+  readonly cwd?: string;
+  readonly debug?: boolean;
+  readonly deep?: number;
+  readonly dot?: boolean;
+  readonly expandDirectories?: boolean;
+  readonly followSymbolicLinks?: boolean;
+  readonly globstar?: boolean;
+  readonly ignore?: string | readonly string[];
+  readonly onlyDirectories?: boolean;
+  readonly onlyFiles?: boolean;
```

Comment:
not sure if adding readonly here is a good idea, mainly because i see no glob library (`node:fs` glob, `glob`, `fast-glob`) other than `globby` doing it 🤔. is it usual for libraries to do this?

Reply:
Wow, that is so odd...

I did some digging and looks like this is the [canonical feature request](https://github.com/microsoft/TypeScript/issues/13347) and a [PR to add a new strict flag for this](https://github.com/microsoft/TypeScript/pull/58296).

Maybe one day we'll have this functionality, but until then I'll remove!